src/core/prototypes/plot_segments/region_2d/polar.ts

```diff
@@ -679,10 +690,21 @@ export class PolarPlotSegment extends PlotSegmentClass<
           }
         },
         type: Specification.AttributeType.Enum,
-        default: "ascending"
+        default: defaultValue
       });
     }
     if (this.object.properties.yData) {
+      const values = (this.object.properties.yData as any).categories;
+      let defaultValue = "ascending";
+      if (values) {
+        const a = values[0].toString();
+        const b = values[(values as any[]).length - 1].toString();
+        if (b && a && b.localeCompare(a) > -1) {
+          defaultValue = "ascending";
+        } else {
+          defaultValue = "descending";
+        }
+      }
```

Comment:
I'd pull this out into a function, like *inferSortDirection* and update all the other to use it, cause they're all basically the same, except they pass in a different set of values.

Also, is `values` guaranteed to be `.length > 0`? If not, that might cause the `values[0]` to crash.

Reply:
Fixed